src/sentry/static/sentry/app/views/settings/projectAlerts/edit.tsx

```diff
@@ -19,45 +22,76 @@ type RouteParams = {
 };
 
 type Props = RouteComponentProps<RouteParams, {}> & {
+  api: Client;
   organization: Organization;
   project: Project;
   hasMetricAlerts: boolean;
-};
+} & AsyncView['props'];
+
+type State = {
+  alertType: string;
+} & AsyncView['state'];
+
+class ProjectAlertsEditor extends AsyncView<Props, State> {
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      alertType: '',
+    };
+  }
+
+  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
+    const {location, params, organization, project} = this.props;
+    if (location.pathname.includes('/alerts/metric-rules/')) {
+      return [
+        ['endpoint', `/organizations/${organization.slug}/alert-rules/${params.ruleId}/`],
+      ];
+    } else {
+      return [
+        [
+          'endpoint',
+          `/projects/${organization.slug}/${project.slug}/rules/${params.ruleId}/`,
+        ],
+      ];
+    }
+  }
```

Comment:
If we're adding a request here, does that mean we're making a duplicate request further down?

Reply:
ah, do you mean further down in the lifecycle of this component?

Reply:
I mean further down in the component tree

Reply:
It does look like we're making a duplicate request here - I'll work on refactoring this

Reply:
refactored this, but not sure this is the approach I should be taking?